src/elasticodbc/dlg_wingui.c

```diff
@@ -43,35 +44,47 @@ const struct authmode *GetCurrentAuthMode(HWND hdlg) {
 }
 
 void SetAuthenticationVisibility(HWND hdlg, const struct authmode *am) {
-    int show_basic = (strcmp(am->authtype_str, AUTHTYPE_BASIC) == 0);
-    ShowWindow(GetDlgItem(hdlg, IDC_USER), show_basic);
-    ShowWindow(GetDlgItem(hdlg, IDC_PASSWORD), show_basic);
-    ShowWindow(GetDlgItem(hdlg, IDC_PASSWORD_STATIC), show_basic);
-    ShowWindow(GetDlgItem(hdlg, IDC_USERNAME_STATIC), show_basic);
-    int show_iam = (strcmp(am->authtype_str, AUTHTYPE_IAM) == 0);
-    ShowWindow(GetDlgItem(hdlg, IDC_REGION), show_iam);
-    ShowWindow(GetDlgItem(hdlg, IDC_REGION_STATIC), show_iam);
+    if (strcmp(am->authtype_str, AUTHTYPE_NONE) == 0) {
```

Comment:
I would suggest doing the logic such that you check for BASIC or IAM, then default to all FALSE if it's not either.

Not sure if the user could ever get an erroneous value in, but it would protect from that.

Reply:
Editing the registry directly might be a case, so I agree with the logic change here

Reply:
changed